oci/table_oci_releasenote.go

```diff
@@ -0,0 +1,242 @@
+// TODO the url is composed of https://docs.oracle.com/en-us/iaas/releasenotes/services/<url encoded name of the sercice>/
+// TODO when the where clause contains a reference to the service, the data can be fetched from this far smaller subset of focused data
+
+func listReleaseNotes(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Debug("listReleaseNotes")
+
+	equalQuals := d.KeyColumnQuals
+	serviceFilter := strings.ToLower(strings.ReplaceAll(equalQuals["service"].GetStringValue(), " ", "-"))
+
+	// TODO: some services have a special mapping to the corresponding URL; for example:
+	// application-performance-monitoring => apm
+	// artifact-registry => artifacts
+	// block-volume => blockvolume
+	if serviceFilter == "application-performance-monitoring" {
+		serviceFilter = "apm"
+	}
+	if serviceFilter == "artifact-registry" {
+		serviceFilter = "artifacts"
+	}
+	if serviceFilter == "block-volume" {
+		serviceFilter = "blockvolume"
+	}
+	if serviceFilter == "file-storage" {
+		serviceFilter = "filestorage"
+	}
+
+	logger.Debug("listReleaseNotes, filter on service " + serviceFilter)
+
+	pagesLeft := true
+	currentPage := 0
+	for pagesLeft {
+		currentPage++
+		releaseNotes, numberOfPages, err := getOCIReleaseNotes(serviceFilter, currentPage)
+		if err != nil {
+			logger.Error("oci_releasenote.listReleaseNotes", "scrape data error", err)
+			return nil, err
+		}
+		for _, releaseNote := range releaseNotes {
+			d.StreamListItem(ctx, releaseNote)
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		// no futher data retrieval when the current page is the last page or when the service filter was defined
+		pagesLeft = currentPage < numberOfPages && serviceFilter == ""
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+// RELEASE_DATE_FORMAT one of two date formats date used in release notes
+const RELEASE_DATE_FORMAT = "Jan. 2, 2006"
+
+// ALTERNATIVE_DATE_FORMAT the other one of the two date formats date used in release notes
+const ALTERNATIVE_DATE_FORMAT = "January 2, 2006"
+
+const ALLOWED_DOMAIN = "docs.oracle.com"
+
+// note: the most recent 50 release notes are published as RSS feed at: https://docs.oracle.com/en-us/iaas/releasenotes/feed/
+// using this feed is (far) less brittle than the screenscraping approach
+
+// the url where the web pages with release notes are available
+const RELEASE_NOTES_BASE_URL = "https://docs.oracle.com/en-us/iaas/releasenotes/"
+const BASE_DOCUMENTATION_URL = "https://docs.oracle.com"
+
+func getOCIReleaseNotes(service string, page int) (releaseNotes []ReleaseNote, numberOfPages int, err error) {
+	c := colly.NewCollector(
+		colly.AllowedDomains(ALLOWED_DOMAIN),
+	)
+	releaseNotes = make([]ReleaseNote, 0, 50)
+	c.OnHTML(".uk-article", func(e *colly.HTMLElement) {
```

Comment:
In our RSS plugin, we use [gofeed](https://github.com/mmcdole/gofeed), which makes parsing RSS for that table much easier. Have you seen this package, would it help you parse the RSS feeds more cleanly?

Reply:
A few example uses in the plugin (we don't manipulate the response data that much, but instead just pass them to columns):
- https://github.com/turbot/steampipe-plugin-rss/blob/315ac92bf5b5fb49bf8ea449ad292d190b354ceb/rss/table_rss_channel.go#L46-L56
- https://github.com/turbot/steampipe-plugin-rss/blob/315ac92bf5b5fb49bf8ea449ad292d190b354ceb/rss/table_rss_item.go#L44-L60